rust/phoenix-sdk-core/src/sdk_client_core.rs

```diff
@@ -421,8 +424,50 @@ impl SDKClientCore {
                             fees_collected_in_quote_lots * self.quote_lot_size,
                         ),
                     }),
+                    PhoenixMarketEvent::TimeInForce(TimeInForceEvent {
+                        index,
+                        order_sequence_number,
+                        last_valid_slot,
+                        last_valid_unix_timestamp_in_seconds,
+                    }) => market_events.push(PhoenixEvent {
+                        market: header.market,
+                        sequence_number: header.sequence_number,
+                        slot: header.slot,
+                        timestamp: header.timestamp,
+                        signature: *sig,
+                        signer: header.signer,
+                        event_index: index as u64,
+                        details: MarketEventDetails::TimeInForce(TimeInForce {
+                            order_sequence_number,
+                            last_valid_slot,
+                            last_valid_unix_timestamp_in_seconds,
+                        }),
+                    }),
+                    PhoenixMarketEvent::ExpiredOrder(ExpiredOrderEvent {
+                        index,
+                        maker_id,
+                        order_sequence_number,
+                        price_in_ticks,
+                        base_lots_removed,
+                    }) => market_events.push(PhoenixEvent {
+                        market: header.market,
+                        sequence_number: header.sequence_number,
+                        slot: header.slot,
+                        timestamp: header.timestamp,
+                        signature: *sig,
+                        signer: header.signer,
+                        event_index: index as u64,
+                        details: MarketEventDetails::Reduce(Reduce {
```

Comment:
@eugene-chen I'm not super happy about this as it merges the TIF cancel into a Reduce. It was the easiest way to get everything working without a major refactor. Happy to change it if necessary

Reply:
tldr on what needs to be refactored? I definitely prefer to maintain the 1:1 mapping between `PhoenixMarketEvent` and `MarketEventDetails`

Reply:
You would need a new enum in MarketEventDetails. Reduce happens to encapsulate the information needed, but it's not super clean. This was the minimum work needed to get testing to work